src/server/handlers.ts

```diff
@@ -0,0 +1,1655 @@
+    }
+
+    let dfxResolver: DfxResolver | undefined;
+    let dfxChangeTimeout: ReturnType<typeof setTimeout>;
+    function notifyDfxChange() {
+        isWorkspaceReady = false;
+        clearTimeout(dfxChangeTimeout);
+        dfxChangeTimeout = setTimeout(async () => {
+            try {
+                dfxResolver = new DfxResolver(() => {
+                    if (!workspaceFolders?.length) {
+                        return null;
+                    }
+                    const folder = workspaceFolders[0];
+                    // for (const folder of workspaceFolders) {
+                    const basePath = resolveFilePath(folder.uri);
+                    const dfxPath = join(basePath, 'dfx.json');
+                    if (existsSync(dfxPath)) {
+                        return dfxPath;
+                    }
+                    return null;
+                    // }
+                });
+
+                const projectDir = await dfxResolver.getProjectDirectory();
+                const dfxConfig = await dfxResolver.getConfig();
+                if (projectDir && dfxConfig) {
+                    if (dfxConfig.canisters) {
+                        try {
+                            const candidPath = join(
+                                projectDir,
+                                '.dfx/local/lsp',
+                            );
+                            const candidUri = URI.file(candidPath).toString();
+
+                            // Add management canister Candid file
+                            const icPath = join(candidPath, 'aaaaa-aa.did');
+                            if (!existsSync(icPath)) {
+                                const icUri = URI.file(icPath).toString();
+                                writeVirtual(
+                                    resolveVirtualPath(icUri),
+                                    icCandid,
+                                );
+                            }
+
+                            const idsPath = join(
+                                projectDir,
+                                '.dfx/local/canister_ids.json',
+                            );
+                            const aliases: Record<string, string> = {};
+                            if (existsSync(idsPath)) {
+                                const canisterIds = JSON.parse(
+                                    readFileSync(idsPath, 'utf8'),
+                                );
+                                Object.entries(canisterIds).forEach(
+                                    ([name, ids]: [string, any]) => {
+                                        const keys = Object.keys(ids);
+                                        // Choose the only principal (or 'local' if multiple are defined)
+                                        const key =
+                                            keys.length === 1
+                                                ? keys[0]
+                                                : 'local';
```

Comment:
This is auto-formating I guess. Is it a standard one?  Do you think that Dfinity developers will have the same setting?

Reply:
Yes. AFAIU it gets reformatted automatically on precommit hook.